app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/OwnedListAdapter.java

```diff
@@ -0,0 +1,83 @@
+package com.cmput301f20t21.bookfriends.ui.library;
+
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.MenuInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.widget.PopupMenu;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.cmput301f20t21.bookfriends.R;
+import com.cmput301f20t21.bookfriends.entities.Book;
+
+import java.util.ArrayList;
+
+public class OwnedListAdapter extends RecyclerView.Adapter<OwnedListAdapter.ViewHolder> {
+    private ArrayList<Book> books;
+
+    public OwnedListAdapter(ArrayList<Book> books) {
+        this.books = books;
+    }
+
+    @NonNull
+    @Override
+    public OwnedListAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_book_list, parent, false);
+        return new OwnedListAdapter.ViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull OwnedListAdapter.ViewHolder holder, int position) {
+        Book book = books.get(position);
+        holder.title.setText(book.getTitle());
+        holder.author.setText(holder.root.getResources().getString(R.string.book_list_item_author, book.getAuthor()));
+        holder.owner.setText("");
+        holder.isbn.setText(book.getIsbn());
+        holder.setBook(book);
+    }
+
+    @Override
+    public int getItemCount() {
+        return books.size();
+    }
+
+    public static class ViewHolder extends RecyclerView.ViewHolder {
+        TextView title;
+        TextView author;
+        TextView owner;
+        TextView isbn;
+        ImageButton moreBtn;
+
+        Book book;
+        View root;
```

Comment:
question: is there a reason that these attributes are all default?

Reply:
https://stackoverflow.com/a/215505/7358099
had no idea about access control when creating them. but looking back on this, I should make them private and extract a binding method for the view holder. For right now, it works, and default is more restricted than public. will make a change.

Reply:
actually I think we should stick to default. I am currently working on abstracting the list adapter and the view holder. having child view holder inheriting the parents' views means that those parent view needs to be at least default. for children to access it, default is the best option imo.